src/pages/BlockDetails.vue

```diff
@@ -0,0 +1,335 @@
+            </BaseLink>
+          </div>
+        </div>
+      </template>
+    </BaseContentBlock>
+  </div>
+
+  <BaseContentBlock
+    :title="$t('Block transactions')"
+    class="transactions-list-page"
+  >
+    <div class="content-row">
+      <TransactionTypeFilter
+        v-model="tab"
+        class="transactions-list-page__tabs"
+      />
+      <TransactionStatusFilter
+        v-model="status"
+        class="transactions-list-page__status"
+      />
+    </div>
+
+    <BaseTable
+      :loading="table.loading.value"
+      :pagination="table.pagination.value"
+      :items="table.items.value"
+      container-class="transactions-list-page__container"
+      @next-page="table.nextPage()"
+      @prev-page="table.prevPage()"
+      @set-page="table.setPage($event)"
+      @set-size="table.setSize($event)"
+    >
+      <template #row="{ item }: { item: Transaction }">
+        <div class="transactions-list-page__row">
+          <TransactionStatus
+            type="tooltip"
+            class="transactions-list-page__icon"
+            :committed="item.committed"
+          />
+
+          <div class="transactions-list-page__column">
+            <div class="transactions-list-page__label">TxID</div>
+
+            <BaseHash
+              :hash="item.hash"
+              :type="hashType"
+              :link="'/transactions/' + item.hash"
+              copy
+            />
+
+            <div class="transactions-list-page__time">
+              {{ format(item.payload.creation_time) }}
+            </div>
+          </div>
+        </div>
+      </template>
+    </BaseTable>
+  </BaseContentBlock>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted } from 'vue';
+import { useRoute } from 'vue-router';
+import { useBlockTable } from '~shared/lib/table';
+import { useWindowSize, computedEager } from '@vueuse/core';
+import { http } from '~shared/api';
+import { TransactionStatus } from '~entities/transaction';
+import { format } from '~shared/lib/time';
+import {
+  filterTransactionsModel as ftm,
+  TransactionStatusFilter,
+  TransactionTypeFilter,
+} from '~features/filter-transactions';
+import invariant from 'tiny-invariant';
+
+import BaseContentBlock from '~base/BaseContentBlock.vue';
+import BaseTable from '~base/BaseTable.vue';
+import BaseHash from '~shared/ui/components/BaseHash.vue';
+import BaseLink from '~shared/ui/components/BaseLink.vue';
+import ArrowIcon from '~icons/arrow.svg';
+
+const WINDOW_WIDTH_BREAKPOINT_TO_SHRINK_HASH = 1200;
+
+const status = ref<ftm.Status>(null);
+const tab = ref<ftm.Tab>('all');
+
+const { width } = useWindowSize();
+
+type HashType = 'full' | 'short'
+
+const hashType = computedEager<HashType>(() =>
+  width.value < WINDOW_WIDTH_BREAKPOINT_TO_SHRINK_HASH ? 'short' : 'full',
+);
+
+const route = useRoute();
+const heightString = route.params.id;
+invariant(typeof heightString === 'string');
+const height = parseInt(heightString, 10);
```

Comment:
Strictly speaking, `route.params.id` is a reactive value. Therefore, `height` should be a computed.

Reply:
Acknowledged and committed 👍🏼
( it would be great if you can point to me a resource where more about these small details can be read. )

Reply:
[Accessing the Router and current Route inside `setup`](https://router.vuejs.org/guide/advanced/composition-api.html#Accessing-the-Router-and-current-Route-inside-setup)